WeatherDispatcher/src/main/java/com/example/WeatherDispatcher/configuration/ConfigureThreadPool.java

```diff
@@ -0,0 +1,17 @@
+package com.example.WeatherDispatcher.configuration;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+
+@Configuration
+public class ConfigureThreadPool {
+
+    @Bean
+    public ThreadPoolTaskScheduler threadPoolTaskScheduler(){
+        ThreadPoolTaskScheduler threadPoolTaskScheduler = new ThreadPoolTaskScheduler();
+        threadPoolTaskScheduler.setPoolSize(Runtime.getRuntime().availableProcessors());
```

Comment:
why to use pool size as the number of processors?

Reply:
done